airflow/api_connexion/endpoints/task_instance_endpoint.py

```diff
@@ -14,46 +14,260 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+from typing import Any, List, Optional, Tuple
+
+from flask import request
+from marshmallow import ValidationError
+from sqlalchemy import and_, func
 
-# TODO(mik-laj): We have to implement it.
-#     Do you want to help? Please look at: https://github.com/apache/airflow/issues/8132
 from airflow.api_connexion import security
+from airflow.api_connexion.exceptions import NotFound, BadRequest
+from airflow.api_connexion.parameters import format_datetime, format_parameters
+from airflow.api_connexion.schemas.task_instance_schema import (
+    clear_task_instance_form,
+    TaskInstanceCollection,
+    task_instance_collection_schema,
+    task_instance_schema,
+    task_instance_batch_form,
+    task_instance_reference_collection_schema,
+    TaskInstanceReferenceCollection,
+)
+from airflow.configuration import conf
+from airflow.models.dagrun import DagRun as DR
+from airflow.models.taskinstance import clear_task_instances, TaskInstance as TI
+from airflow.models.dag import DagModel
+from airflow.models.dagbag import DagBag
+from airflow.models import SlaMiss
+from airflow.utils.state import State
+from airflow.utils.session import provide_session
 
 
-@security.requires_access([("can_read", "Dag"), ("can_read", "DagRun"), ("can_read", "Task")])
-def get_task_instance():
-    """
-    Get a task instance
-    """
-    raise NotImplementedError("Not implemented yet.")
+@security.requires_access(
+    [
+        ("can_read", "Dag"),
+        ("can_read", "DagRun"),
+        ('can_read', 'Task'),
+    ]
+)
+@provide_session
+def get_task_instance(dag_id: str, dag_run_id: str, task_id: str, session=None):
+    """Get task instance"""
+    query = (
+        session.query(TI)
+        .filter(TI.dag_id == dag_id)
+        .join(DR, and_(TI.dag_id == DR.dag_id, TI.execution_date == DR.execution_date))
```

Comment:
```suggestion
        .join(DR)
```

should work because of https://github.com/apache/airflow/blob/c47383dffff1d216d1c4985b4a9cc7a7310a837e/airflow/models/dagrun.py#L77-L82

Reply:
Unfortunately it doesn't work. I have the following message.
```
sqlalchemy.exc.InvalidRequestError: Don't know how to join to <class 'airflow.models.dagrun.DagRun'>. Please use the .select_from() method to establish an explicit left side, as well as providing an explcit ON clause if not present already to help resolve the ambiguity.
```